[Feature] Timer UI 및 기본 동작 구현

Peter1119 opened the pull request:
## 🔗 관련 이슈
<!-- 이 PR이 연결된 이슈 번호를 명시하세요 (예: Closes #123, Resolves #45) -->
- 관련 이슈: #

---

## ✨ 작업 내용
<!-- 어떤 기능/버그/리팩토링 작업을 했는지 간단하게 작성해주세요 -->
-  MVI 패턴으로 View 및 기본적인 Timer 동작 확인
-  통합 후 백그라운드 및 동기화 테스트 해볼 예정입니다. 
- 

---

## 📸 Showcase
<!-- UI 변경이 있는 경우, 캡처 이미지나 GIF를 첨부해주세요 -->
| 변경 전 | 변경 후 |
|--------|--------|
| 이미지 | 이미지 |
<img width="549" height="1140" alt="image" src="https://github.com/user-attachments/assets/ce03decb-c78f-4809-a2fa-e04037490e60" />
<img width="534" height="1139" alt="image" src="https://github.com/user-attachments/assets/ed23af71-d279-49cc-804d-79f9d0eb86cd" />
<img width="543" height="1137" alt="image" src="https://github.com/user-attachments/assets/8841921e-3715-444b-9d66-ffe17637ac25" />

> 📌 이미지가 없다면 이 섹션은 생략해도 됩니다.

---

## 📝 참고 사항
<!-- 리뷰어가 이해를 돕기 위한 추가 설명이나 논의 사항이 있다면 여기에 작성하세요 -->
- 

## Motivation 🥳 (코드를 추가/변경하게 된 이유)

## Key Changes 🔥 (주요 구현/변경 사항)

## To Reviewers 🙏 (리뷰어에게 전달하고 싶은 말)

## Reference 🔗


## Close Issues 🔒 (닫을 Issue)
Close #No.

## Checklist
- [ ] 브랜치를 가져와 작업한 경우 이전 브랜치에 PR을 보냈는지 확인
- [ ] 빌드를 위해 SceneDelegate 수정한 것 PR로 올리지 않았는지 확인
- [ ] 필요없는 주석, 프린트문 제거했는지 확인
- [ ] 컨벤션 지켰는지 확인
- [ ] final, private 제대로 넣었는지 확인
- [ ] 다양한 디바이스에 레이아웃이 대응되는지 확인
  - [ ] iPhone SE
  - [ ] iPhone 13
  - [ ] iPhone 13 Pro Max

Peter1119 committed:
init: Timer 모듈 생성

Peter1119 committed:
feat: MVI 아키텍처 기반 Timer 모델 레이어 구현

- TimerIntent: 사용자 액션 정의 (시간 선택, 타이머 제어)
- TimerState: UI 상태 관리 및 계산 속성 구현
- TimerSideEffect: 비동기 작업을 위한 side effect 정의
- TimerModel: reduce 함수 기반 비즈니스 로직 처리

Peter1119 committed:
feat: Timer UI 컴포넌트 구현

- CircularProgressView: 원형 프로그레스 바 및 시간 표시
- TimePickerView: 시/분/초 선택 휠 피커
- TimerControlButtons: 시작/취소/일시정지/재개 버튼

Peter1119 committed:
feat: Timer ViewModel 및 메인 View 구현

- TimerContainer: MVI 패턴 기반 ViewModel, Combine 활용한 타이머 로직
- TimerView: 컴포넌트 조합한 메인 타이머 화면
- 0.3초 지연 후 타이머 시작으로 애니메이션 동기화

Peter1119 committed:
chore: 기존 Base.swift 제거 및 프로젝트 파일 업데이트

- Base.swift 템플릿 파일 제거
- Xcode 프로젝트에 새로운 소스 파일들 반영

minneee reviewed (approved):
수고하셨습니다!

Roy-wonji reviewed (approved):
@Peter1119  수고 하셨습니다

Peter1119 committed:
chore: main 브랜치 병합 및 충돌 해결

- Alarm, StopWatch, Widget 모듈 추가
- Timer 모듈과 공존하도록 Workspace 구성
- Utill 모듈에 Extension 추가
- Presentation 모듈 Sources 구조 업데이트

Peter1119 committed:
chore: Presentation 프로젝트 Sources 그룹 구조 업데이트

- PresntaionExport.swift를 위한 Sources/Exported 그룹 추가
- tuist generate 결과 반영
